docs/Fees.md

```diff
@@ -0,0 +1,106 @@
+# Fees
+
+The [`Bridge.sol`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/Bridge.sol) contract collects fees on every new deposit made. To achieve this, it calls a fee handler specified using the [`adminChangeFeeHandler`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/Bridge.sol#L202) method. Different fee handler implementations can be used as long as they follow the [`IFeeHandler`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/interfaces/IFeeHandler.sol) interface.
+
+> In all Sygma environments, the [`FeeHandlerRouter`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/handlers/FeeHandlerRouter.sol) implementation is used as this root `IFeeHandler`. This particular implementation allows for more granular control of handling fees based on the resourceID and destination domainID.
+
+## IFeeHandler interface
+
+The [`IFeeHandler`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/interfaces/IFeeHandler.sol interface outlines the events and function signatures required by any fee handler implementation. The following function signatures are mandatory:
+
+- [`collectFee(...)`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/interfaces/IFeeHandler.sol#L50) - This function collects the fee based on the parameters provided and is called internally on every deposit request.
+- [`calculateFee(...)`](https://github.com/sygmaprotocol/sygma-solidity/blob/master/contracts/interfaces/IFeeHandler.sol#L63) - This function calculates the final fee amount based on the given parameters. It returns the final fee amount as a `uint256` and the address of the token in which the fee is being paid as an `address`.
+
+### FeeCollected event
+
+The fee handler implementation should emit the `FeeCollected` event every time the fee is successfully collected on the source chain. The `tokenAddress` parameter represents the token address in which the fee was collected, with `0` representing the base currency.
```

Comment:
Actually `FeeCollected` will also be emitted with zeroAddress when `DynamicGenericFeeHandlerEVM` calculates the fee. 
With that on mind some kind of disclaimer could be useful for tracking transactions on chain.